src/Tasks.UnitTests/TaskFactoryUtilities_Tests.cs

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using Microsoft.Build.Shared;
+using Microsoft.Build.Shared.FileSystem;
+using Microsoft.Build.Utilities;
+using Shouldly;
+using Xunit;
+
+#nullable disable
+
+namespace Microsoft.Build.UnitTests
+{
+    public sealed class TaskFactoryUtilities_Tests
+    {
+        [Fact]
+        public void CreateProcessSpecificTaskDirectory_ShouldCreateValidDirectory()
+        {
+            // Act
+            string directory = TaskFactoryUtilities.CreateProcessSpecificTaskDirectory();
+
+            // Assert
+            directory.ShouldNotBeNull();
+            Directory.Exists(directory).ShouldBeTrue();
+            directory.ShouldContain(TaskFactoryUtilities.InlineTaskTempDllSubPath);
+            directory.ShouldContain($"pid_{EnvironmentUtilities.CurrentProcessId}");
```

Comment:
nit: I feel that this line might be getting on the border of checking the implementation details that could be changed. I would probably avoid that in the tests.